File: luogu/luogu-P1200.cpp
#include <iostream>
#include <string>
using namespace std;
int main(){
    string a;
    string b;
    cin >> a;
    cin >> b;
    int an = 1,bn = 1;
    for (int i = 0; i < a.size(); ++i) {
        an *= (a[i] - 'A' + 1);
    }
    for (int i = 0; i < b.size(); ++i) {
        bn *= (b[i] - 'A' + 1);
    }
    if ((an % 47) == (bn % 47)){
        cout << "GO" << endl;
    }
    else{
        cout << "STAY" << endl;
    }
    return 0;
}

File: luogu/luogu-P1200.py
def get_num(id):
    num = 1
    for i in id :
        num *= (leters.index(i.lower())+1)
    return num % 47
leters = 'abcdefghijklmnopqrstuvwxyz'
id1 = input()
id2 = input()
if get_num(id1) == get_num(id2) :
    print('GO')
else :
    print('STAY')
